lib/resource-mapper.js

```diff
@@ -90,6 +90,14 @@ class ResourceMapper {
     return { path, contentType: contentType || this._defaultContentType }
   }
 
+  async _getMatchingFile (folder, filename, isIndex) {
+    const files = await this._readdir(folder)
+
+    // Find a file with the same name (minus the dollar extension)
+    return (files.find(f => this._removeDollarExtension(f) === filename ||
+      (isIndex && f.startsWith(this._indexFilename + '.') && !f.endsWith(this._suffixes.acl))))
```

Comment:
I would go one step further, and loop over all `this._suffixes` to exclude _all_ file names that end with any of the defined suffixes. Right now, only `index.html.acl` will be blacklisted, but `index.html.meta` will still cause the same problem. A loop over `this._suffixes` would solve this.

For this purpose, it may be easier to refactor `this._suffixes` to be an array instead of a hash, as this is (AFAICS) only being used here.